File: java/PWMNavXMecanum/src/main/java/frc/robot/Robot.java
package frc.robot;

import com.kauailabs.navx.frc.AHRS;

import edu.wpi.first.wpilibj.Joystick;
import edu.wpi.first.wpilibj.PWMVictorSPX;
import edu.wpi.first.wpilibj.TimedRobot;
import edu.wpi.first.wpilibj.I2C.Port;
import edu.wpi.first.wpilibj.drive.MecanumDrive;

/**
 * This is a sample program that uses mecanum drive with the VMX-Pi integrated NavX sensor to
 * maintain rotation vectors in relation to the starting orientation of the robot
 * (field-oriented controls).
 */
public class Robot extends TimedRobot {

  private MecanumDrive m_robotDrive;
  private final AHRS m_gyro = new AHRS(Port.kMXP);
  private final Joystick m_joystick = new Joystick(0);

  @Override
  public void robotInit() {
    PWMVictorSPX frontLeft = new PWMVictorSPX(0);
    PWMVictorSPX rearLeft = new PWMVictorSPX(1);
    PWMVictorSPX frontRight = new PWMVictorSPX(2);
    PWMVictorSPX rearRight = new PWMVictorSPX(3);

    

    // Invert the left side motors.
    // You may need to change or remove this to match your robot.
    frontLeft.setInverted(true);
    rearLeft.setInverted(true);

    m_robotDrive = new MecanumDrive(frontLeft, rearLeft, frontRight, rearRight);

  }

  /**
   * Mecanum drive is used with the gyro yaw as an input.
   */
  @Override
  public void teleopPeriodic() {
    m_robotDrive.driveCartesian(m_joystick.getX(), m_joystick.getY(),
        m_joystick.getZ(), m_gyro.getYaw());
  }
}

File: cpp/PWMNavXMecanum/src/main/cpp/Robot.cpp
#include "AHRS.h"
#include <frc/SPI.h>
#include <frc/Joystick.h>
#include <frc/PWMVictorSPX.h>
#include <frc/TimedRobot.h>
#include <frc/drive/MecanumDrive.h>

/**
 * This is a sample program that uses mecanum drive with the VMX-Pi integrated NavX sensor to
 * maintain rotation vectors in relation to the starting orientation of the robot
 * (field-oriented controls).
 */
class Robot : public frc::TimedRobot {
 public:
  void RobotInit() override {
    // Invert the left side motors. You may need to change or remove this to
    // match your robot.
    m_frontLeft.SetInverted(true);
    m_rearLeft.SetInverted(true);

  }

  /**
   * Mecanum drive is used with the gyro yaw as an input.
   */
  void TeleopPeriodic() override {
    m_robotDrive.DriveCartesian(m_joystick.GetX(), m_joystick.GetY(),
                                m_joystick.GetZ(), m_gyro.GetYaw());
  }

 private:

  frc::PWMVictorSPX m_frontLeft{0};
  frc::PWMVictorSPX m_rearLeft{1};
  frc::PWMVictorSPX m_frontRight{2};
  frc::PWMVictorSPX m_rearRight{3};
  frc::MecanumDrive m_robotDrive{m_frontLeft, m_rearLeft, m_frontRight, m_rearRight};

  AHRS m_gyro{frc::SPI::Port::kMXP}; //Uses the VMX-Pi on-board navX sensors
  frc::Joystick m_joystick{0};
};

#ifndef RUNNING_FRC_TESTS
int main() { return frc::StartRobot<Robot>(); }
#endif
